### PR TITLE
feat: Localization

### PR DESCRIPTION
**Pull request by scottyg**
@deer I've set up basic localization, this should solve #14 and #15. 

It currently only has English and Spanish but we can add more languages at `./src/utils/localization`. There is also a switch statement in `./src/plugin/blog.ts` that checks for the lang set in blogOptions. If someone wants to customize their own words they can add a second parameter to blogPlugin(). You can see this working at `./tests/localization_fixture`. I added some instructions to the read me but I feel like they could be more clear.

Let me know if this works. Cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,16 +59,33 @@ const blogOptions: BlogOptions = {
...
+  attribution: "By",
+  nextPage: "Next Page →",
+  previousPage: "← Previous Page",
+  nextPost: "Next Post →",
+  previousPost: "← Previous Post",
+  continueReading: "Continue reading →",
+  noPostsFound: "No posts found. Start writing!",
+  blogTitleEnding: " — Blog",
+  archiveTitleEnding: " — Archive",
+  authorTitleEnding: " — Author Archive",
```
Comment (deer) [inline README.md:81]: I'm open to suggestions on how to improve this default as well. But I guess it's fine for now... (And yes, I'm aware that these are my original defaults, I just was never fully satisfied with them!)

File: src/utils/localization/es.ts
Comment (deer) [inline src/utils/localization/es.ts:1]: ¿Qué bien hablas español? I speak well enough Spanish to read these and understand, but I can't judge how colloquial/natural the translations are. But I suppose it's irrelevant -- looks good to me, and I suppose that's good enough until someone complains!

File: tests/localization_fixture/deno.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "lock": false,
+  "tasks": {
+    "start": "deno run -A --watch=static/,routes/ dev.ts",
+    "update": "deno run -A -r https://fresh.deno.dev/update ."
+  },
+  "imports": {
+    "$fresh/": "https://raw.githubusercontent.com/deer/fresh/feat_plugin_islands/",
```
Comment (deer) [inline tests/localization_fixture/deno.json:8]: I hope to remove all this nonsense once https://github.com/denoland/fresh/pull/1472 gets merged. Then we can just use the standard version.

File: tests/localization_fixture/options.ts
Comment (deer) [inline tests/localization_fixture/options.ts:1]: Did this sneak in? It doesn't seem to be used.

**Review by deer — COMMENTED** [review]
Great, thanks for opening this. A few thoughts:
1. Where are the actual tests? I don't see anything actually running.
2. Do we need so many posts? Perhaps we could get away with... two posts? Then setting `postsPerPage` to 1 would force a 'next page' and 'previous page' to appear at least somewhere. Two posts would also enable 'next post' and 'previous post'.
3. Why did you decide to make `Localization` independent from `BlogOptions`? It seems like you could instead have...
```tsx
interface BlogOptions {
  title: string;
  navbarItems: Record<string, string>;
  rootPath: string;
  postsPerPage?: number;
  sources?: Source[];
  useSeparateIndex?: boolean;
  markdownStyle?: "light" | "dark" | "auto";
  strings?: Localization | Partial<Strings>;
}

type Strings = {
  attribution: string;
  nextPage: string;
  previousPage: string;
  nextPost: string;
  previousPost: string;
  continueReading: string;
  noPostsFound: string;
  blogTitleEnding: string;
  archiveTitleEnding: string;
  authorTitleEnding: string;
};

type Localization = Strings & { lang: Languages };

type Languages = "en" | "es" | "de";

export const en: Localization = {
  lang: "en",
  attribution: "By",
  nextPage: "Next Page →",
  previousPage: "← Previous Page",
  nextPost: "Next Post →",
  previousPost: "← Previous Post",
  continueReading: "Continue reading →",
  noPostsFound: "No posts found. Start writing!",
  blogTitleEnding: " — Blog",
  archiveTitleEnding: " — Archive",
  authorTitleEnding: " — Author Archive",
};

export const es: Localization = {
  lang: "es",
  attribution: "Por",
  nextPage: "Página siguiente →",
  previousPage: "← Página anterior",
  nextPost: "Entrada Siguiente →",
  previousPost: "← Entrada Anterior",
  continueReading: "Seguir leyendo →",
  noPostsFound: "No se han encontrado publicaciones. ¡Empieza a escribir!",
  blogTitleEnding: " — Blog",
  archiveTitleEnding: " — Archivo",
  authorTitleEnding: " — Archivo del autor",
};
```
This isn't fully thought out and should definitely be split into multiple files (like you have), but the use case I'm trying to consider is what if someone is happy with the default words, but really dislikes the arrows? The only thing they want to do is override `nextPage`, `previousPage`, `nextPost`, and `previousPost` with strings that don't have arrows. There would need to be some sort of object merging, where we take the default localization, and then apply the partial on top of it to override the properties.
4. Looks like you're getting rekt by CI. I recommend running `deno task check` locally to catch those formatting and linting issues.

Thanks again for the PR. Please let me know if you don't have time to address these comments. (Or if you don't want to!)

File: src/utils/localization/es.ts
Comment (scottyg) [reply src/utils/localization/es.ts:1]: Not very well, I just used online translation so I could test out functionality. My guess is that its not perfect.

File: README.md (hunk 1 above)
Comment (scottyg) [reply README.md:81]: I think this is good for now. Its straight to the point and users can always override them if they want.

**Comment by scottyg** [comment]
@deer This was lazy Sunday coding. I've implemented all of the suggestions mentioned above. Last thing I need to do is figure out a better way to switch the languages with out using a switch statement. That section of code could get quite large as the project grows. I'll also create some tests and do another PR after work today.

**Comment by deer** [comment]
Cool, looking forward to the next iteration! Just out of curiosity: why close this one and open another instead of just adding to this?

**Comment by scottyg** [comment]
@deer Not sure why I did that, lol. I'll reopen to keep things clean.

**Comment by scottyg** [comment]
@deer 🤞 I did better this round

**Comment by deer** [comment]
Looks like I need to sort something out with notion. I will look more properly at this later today. Thanks for following up!

**Comment by scottyg** [comment]
Let me know if I can help.

**Comment by deer** [comment]
Google to the rescue: https://github.com/orgs/community/discussions/26163
What do you think is the most expedient way to resolve this? A few options:
1. I can promote you to a collaborator and then you could push against the repo.
2. I can cherry pick the commit from your fork and push to the repo.
3. As a longer term fix, something about refactoring the tests to isolate notion so that these run less frequently.
4. I'll look to see if there's an easy way to make a notion database public, so that there's no need to authenticate.
5. The shortest term fix is that I'll pull your branch and run the tests locally, then just go ahead and merge after I complete the review.

Starting to review the actual changes now.

File: src/plugin/blog.ts [hunk 3]
```diff
@@ -37,6 +57,9 @@ export function blogPlugin(
       `The specified posts directory '${postsDir}' does not exist.`,
     );
   }
+  const lang: string = options.strings?.lang || "en";
+  const localization: Localization = { ...languages[lang], ...options.strings };
```
Comment (deer) [inline src/plugin/blog.ts:61]: This is great!

File: src/plugin/blog.ts [hunk 4]
```diff
@@ -19,8 +20,27 @@ interface BlogOptions {
...
+  previousPost?: string;
+  continueReading?: string;
+  noPostsFound?: string;
+  blogTitleEnding?: string;
+  archiveTitleEnding?: string;
+  authorTitleEnding?: string;
+  lang?: Languages;
+};
+
+// type Localization = Strings & { lang: Languages };
```
Comment (deer) [inline src/plugin/blog.ts:40]: 🧐

File: src/utils/localization/de.ts
Comment (deer) [inline src/utils/localization/de.ts:1]: I'm not certain about these translations 😂 
Maybe delete these for now and I can add it in later?

**Review by deer — APPROVED** [review]
This is really nice! Thanks so much for contributing. A few really minor comments, and then a suggestion if you want to go the extra mile. Not sure how up-to-date you stay with the latest fresh releases, but in 1.4 there's the possibility to set the lang attribute in the html tag much easier. See https://deno.com/blog/fresh-1.4#custom-html-head-and-body-tags for details. I guess the lang property you've introduced would be perfect for driving this. Of course I can always log an issue and do this later. After your great work here it should be very easy to implement!

I'll approve this now, and then if you address the minor comments I can merge tomorrow morning when I wake up.

**Comment by scottyg** [comment]
> Google to the rescue: https://github.com/orgs/community/discussions/26163 What do you think is the most expedient way to resolve this? A few options:
> 
> 1. I can promote you to a collaborator and then you could push against the repo.
> 2. I can cherry pick the commit from your fork and push to the repo.
> 3. As a longer term fix, something about refactoring the tests to isolate notion so that these run less frequently.
> 4. I'll look to see if there's an easy way to make a notion database public, so that there's no need to authenticate.
> 5. The shortest term fix is that I'll pull your branch and run the tests locally, then just go ahead and merge after I complete the review.

However you want to go about this works for me. I definitely want to help with this project going forward so let me know how I can help. I've never used Notion before so it would take me a second to get used to it. 

> Ok, obviously I'm still a bit new to the open source thing, since the way I've set it up has made it impossible for you to get a passing run on your machine. I'll have a think about that, but in the meantime, your changes are passing just fine locally for me.

I'm new to contributing to open source as well. Good to know were in the same boat.

> This is really nice! Thanks so much for contributing. A few really minor comments, and then a suggestion if you want to go the extra mile. Not sure how up-to-date you stay with the latest fresh releases, but in 1.4 there's the possibility to set the lang attribute in the html tag much easier. See https://deno.com/blog/fresh-1.4#custom-html-head-and-body-tags for details. I guess the lang property you've introduced would be perfect for driving this. Of course I can always log an issue and do this later. After your great work here it should be very easy to implement!

Thats a great idea, I'll work on it today after work. I see `"$fresh/": "https://raw.githubusercontent.com/deer/fresh/feat_plugin_islands/",` in deno.json shouldn't this be the default $fresh import for 1.4?

**Comment by scottyg** [comment]
Just did some clean up of code.

**Comment by deer** [comment]
Oh, one other fyi for the future (in the hope that you continue contributing... 🙏 ): https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

So you can write `closes #14 , closes #15` (or pick your favorite word from the list in the link), and then the issues get closed when the PR is merged.

**Comment by scottyg** [comment]
Good to know, thanks!

**Pull request merged**